geomstats/geometry/hermitian_matrices.py

```diff
@@ -0,0 +1,248 @@
+        """
+        if not gs.all(Matrices.is_hermitian(mat)):
+            logging.warning("non-Hermitian matrix encountered.")
+        mat = Matrices.to_hermitian(mat)
+        return gs.triu_to_vec(mat)
+
+    @staticmethod
+    @geomstats.vectorization.decorator(["vector", "else"])
+    def from_vector(vec, dtype=gs.complex128):
+        """Convert a vector into a Hermitian matrix.
+
+        Parameters
+        ----------
+        vec : array-like, shape=[..., n(n+1)/2]
+            Vector.
+        dtype : dtype, {gs.complex64, gs.complex128}
+            Data type object to use for the output.
+            Optional. Default: gs.complex128.
+
+        Returns
+        -------
+        mat : array-like, shape=[..., n, n]
+            Hermitian matrix.
+        """
+        vec_dim = vec.shape[-1]
+        mat_dim = (gs.sqrt(8.0 * vec_dim + 1) - 1) / 2
+        if mat_dim != int(mat_dim):
+            raise ValueError(
+                "Invalid input dimension, it must be of the form"
+                "(n_samples, n * (n + 1) / 2)"
+            )
+        mat_dim = int(mat_dim)
+        shape = (mat_dim, mat_dim)
+        mask = 2 * gs.ones(shape) - gs.eye(mat_dim)
+        indices = list(zip(*gs.triu_indices(mat_dim)))
+        vec = gs.cast(vec, dtype)
+        upper_triangular = gs.stack(
+            [gs.array_from_sparse(indices, data, shape) for data in vec]
+        )
+        mat = Matrices.to_hermitian(upper_triangular) * gs.cast(mask, dtype)
+        return mat
+
+    @classmethod
+    def expm(cls, mat):
+        """
+        Compute the matrix exponential for a Hermitian matrix.
+
+        Parameters
+        ----------
+        mat : array_like, shape=[..., n, n]
+            Hermitian matrix.
+
+        Returns
+        -------
+        exponential : array_like, shape=[..., n, n]
+            Exponential of mat.
+        """
+        n = mat.shape[-1]
+        dim_3_mat = gs.reshape(mat, [-1, n, n])
+        expm = cls.apply_func_to_eigvals(dim_3_mat, gs.exp)
+        expm = gs.reshape(expm, mat.shape)
+        return expm
+
+    @classmethod
+    def powerm(cls, mat, power):
+        """
+        Compute the matrix power.
+
+        Parameters
+        ----------
+        mat : array_like, shape=[..., n, n]
+            Hermitian matrix with non-negative eigenvalues.
+        power : float, list
+            Power at which mat will be raised. If a list of powers is passed,
+            a list of results will be returned.
+
+        Returns
+        -------
+        powerm : array_like, shape=[..., n, n]
+            Matrix power of mat.
+        """
+        if isinstance(power, list):
+            power_ = [lambda ev, p=p: gs.power(ev, p) for p in power]
+        else:
+
+            def power_(ev):
+                return gs.power(ev, power)
+
+        return cls.apply_func_to_eigvals(mat, power_, check_positive=False)
+
+    @staticmethod
+    def apply_func_to_eigvals(mat, function, check_positive=False):
```

Comment:
This method is basically the same as `SymmetricMatrices.apply_func_to_eigvals`. I think we should generalize it (probably we need to pass the `transpose` method as input) to avoid code duplication.

Reply:
I have not changed this code for now. Tell me if you see a nice way to avoid code duplication. (I can try to pass the transpose method as input if you think that it is the best way to do it.)